workers/ohsome_quality_analyst/indicators/mapping_saturation/indicator.py

```diff
@@ -220,3 +222,30 @@ def select_models(self, fitted_models) -> list:
                 fitted_models.remove(fm)
 
         return fitted_models
+
+
+def validate_query_results(result: dict) -> None:
+    """Validate query results.
+
+    Validation is needed if the input Layer to this Indicator has data attached to it.
+
+    Raises:
+        LayerDataSchemaError: Error based on SchemaError with additional description.
+    """
+    try:
+        Schema(
+            {
+                "result": [
+                    {
+                        "value": Or(float, int),
+                        "timestamp": Use(lambda t: isoparse(t)),
+                    }
+                ]
+            },
+            ignore_extra_keys=True,
+        ).validate(result)
+    except SchemaError as error:
+        raise LayerDataSchemaError(
+            "Invalid Layer data input to the Mapping Saturation Indicator.",
+            error,
+        )
```

Comment:
Maybe move the check to the ohsome client to ensure that the ohsome API responses are valid.

Reply:
Moved the validation function to the ohsome-client module